### PR TITLE
Fixed #27271 - Made ServerFormatter able to handle simple messages

### PR DESCRIPTION
**Pull request by GeyseR**
After this fix ServerFormatter will be able to format simple string
messages or messages with formatting arguments. Also it will set
server_time varible to log record created time if it is required by
format string and wasn't passed in extra parameters

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

File: tests/logging_tests/tests.py [hunk 1]
```diff
@@ -513,3 +516,60 @@ def test_extra_args(self):
...
+        with patch_django_server_logger() as logger_output:
+            logger.info(log_msg, extra={'server_time': server_time})
+            self.assertEqual(
+                '[%s] %s\n' % (server_time, log_msg),
+                logger_output.getvalue()
+            )
+
+        with patch_django_server_logger() as logger_output:
+            logger.info(log_msg)
+            self.assertRegexpMatches(
```
Comment (timgraham) [inline tests/logging_tests/tests.py]: Use `six.assertRegex` to avoid the deprecated alias on Python 3.

File: tests/logging_tests/tests.py [hunk 2]
```diff
@@ -513,3 +516,60 @@ def test_extra_args(self):
...
+                formatter.format(record),
+                getattr(color_style, style)(log_msg),
+            )
+        record = logging.makeLogRecord({'msg': log_msg})
+        self.assertEqual(formatter.format(record), log_msg)
+
+    def test_server_formatter_default_format(self):
+        server_time = '2016-09-25 10:20:30'
+        log_msg = 'log message'
+
```
Comment (timgraham) [inline tests/logging_tests/tests.py]: chop blank line

File: tests/logging_tests/tests.py [hunk 3]
```diff
@@ -513,3 +516,60 @@ def test_extra_args(self):
...
             )]
         )
+
+
+class LogFormattersTests(SetupDefaultLoggingMixin, SimpleTestCase):
+    def test_server_formatter_styles(self):
+        color_style = color.make_style('')
+        formatter = ServerFormatter()
+        formatter.style = color_style
+
```
Comment (timgraham) [inline tests/logging_tests/tests.py]: chop blank line

File: tests/logging_tests/tests.py [hunk 4]
```diff
@@ -513,3 +516,60 @@ def test_extra_args(self):
                 }},
             )]
         )
+
+
+class LogFormattersTests(SetupDefaultLoggingMixin, SimpleTestCase):
+    def test_server_formatter_styles(self):
```
Comment (timgraham) [inline tests/logging_tests/tests.py]: I guess most test classes have a blank line after the class and before the first test method.

File: tests/logging_tests/tests.py [hunk 5]
```diff
@@ -513,3 +516,60 @@ def test_extra_args(self):
                 }},
             )]
         )
+
+
+class LogFormattersTests(SetupDefaultLoggingMixin, SimpleTestCase):
```
Comment (timgraham) [inline tests/logging_tests/tests.py]: Is `SetupDefaultLoggingMixin` needed? Tests seem to be passing without it.

File: django/utils/log.py [hunk 6]
```diff
@@ -164,28 +164,32 @@ def __init__(self, *args, **kwargs):
...
+                # Any 5XX, or any other status code
                 msg = self.style.HTTP_SERVER_ERROR(msg)
 
+        if self.usesServerTime() and not hasattr(record, 'server_time'):
+            record.server_time = self.formatTime(record, self.datefmt)
+
         record.msg = msg
         return super(ServerFormatter, self).format(record)
+
+    def usesServerTime(self):
```
Comment (GeyseR) [inline django/utils/log.py]: Yeah, it looks weird for me, too. But i followed camel case notation from parent `logging.Formatter.usesTime` method. I'm totally ok with changing it to `uses_server_time` if you want. Especially it is supposed to use only in our code.

File: tests/logging_tests/tests.py (hunk 5 above)
Comment (GeyseR) [inline tests/logging_tests/tests.py]: You are right, i'll remove it

File: tests/logging_tests/tests.py (hunk 4 above)
Comment (GeyseR) [inline tests/logging_tests/tests.py]: Yes, except `LoggingFiltersTest` and `CallbackFilterTest` in same module. What do you think about fixing formatting in this test cases, or it should be done in separate commit?

Comment (timgraham) [inline tests/logging_tests/tests.py]: I don't think it's worth changing elsewhere.

**Comment by timgraham** [comment]
merged in 6709ea4ae91b906742506ac0c42a3a272991001f, thanks!
